### PR TITLE
Add support to use env vars in RpcWorkerConfig

### LINKED ISSUE #6179: Remove Java specific environment variables look up from Rpc Worker Config parsing
cc @yojagad 

Following code specifically looks up environment variables for Java worker:

https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs#L216

https://github.com/Azure/azure-functions-host/blob/dev/src/WebJobs.Script/Workers/Rpc/Configuration/RpcWorkerConfigFactory.cs#L232

PR #6064 added ability to expand environment variables for Custom Handlers. Rpc workers can use the functionality and specify environment variables in `worker.config.json`

### PR DESCRIPTION
**Pull request by pragnagopa**
Fixes #6179

### CODE REVIEW

**Comment by pragnagopa** [comment]
@divyagandhii / @vrdmr   - Can you please confirm if `Java_Home` environment variable is set in linux images for consumption?

@ankitkumarr - Can you please confirm if `Java_Home` environment variable is set in linux images for dedicated and elastic premium?

**Comment by divyagandhisethi** [comment]
> 
> 
> @divyagandhii / @vrdmr - Can you please confirm if `Java_Home` environment variable is set in linux images for consumption?
> 
> @ankitkumarr - Can you please confirm if `Java_Home` environment variable is set in linux images for dedicated and elastic premium?

Yes it's set for linux consumption

**Comment by ankitkumarr** [comment]
@pragnagopa, yeah looks like `Java_home` is set for all files similar to [here](https://github.com/Azure/azure-functions-docker/blob/b8783f2e7bf3529d6a76c4c05ca2d286b766eaf6/host/3.0/buster/amd64/java/java11/java11-appservice.Dockerfile#L46).

cc: @amamounelsayed, who created these.

**Comment by ankitkumarr** [comment]
Sorry, I clicked "Close and comment" accidentally.  Re-opened.

**Comment by pragnagopa** [comment]
Thanks @divyagandhii and @ankitkumarr for confirmation.

@amamounelsayed - Green run on the tests! Lets sync up offline on what is best way to get this merged.

File: src/WebJobs.Script/Workers/ProcessManagement/DefaultWorkerProcessFactory.cs [hunk 1]
```diff
@@ -43,6 +55,7 @@ public virtual Process CreateWorkerProcess(WorkerContext context)
                     startInfo.EnvironmentVariables[envVar.Key] = envVar.Value;
                     startInfo.Arguments = startInfo.Arguments.Replace($"%{envVar.Key}%", envVar.Value);
                 }
+                startInfo.Arguments = SanitizeExpandedArgument(startInfo.Arguments);
```
Comment (amamounelsayed) [inline src/WebJobs.Script/Workers/ProcessManagement/DefaultWorkerProcessFactory.cs:58]: Is this mean we are overriding the Arguments? Should it be add?

Comment (pragnagopa) [reply src/WebJobs.Script/Workers/ProcessManagement/DefaultWorkerProcessFactory.cs:58]: After building the list of arguments, this method removes any arguments with format %var% as that variable is not set. For example, java worker config specifies `%JAVA_OPTS%`. If `Java_Opts` is set then arguments will be appended.

**Review by amamounelsayed — APPROVED** [review]

**Comment by amamounelsayed** [comment]
Thank you @pragnagopa  and @mhoeger  for your help :)

**Pull request merged**